### ISSUE TITLE
Healthie: "Get patient" should try and parse the phone number to valid phone data point

### ISSUE THREAD

**Issue by nckhell** [awell-health/awell-extensions#81]
# Context

<img width="910" alt="Screenshot 2023-04-17 at 15 45 32" src="https://user-images.githubusercontent.com/5594064/232502849-7f8994d7-5f7d-45b9-b888-79f603de820c.png">

- https://awellhealth.atlassian.net/wiki/spaces/AWELL/pages/3492773961/Have+5+different+Extensions+implemented+in+a+live+care+flow
- https://awellhealth.slack.com/archives/C04F5459Y56/p1681737233645069

# Situation today

The "Get patient" action ingests new data points in a care flow, one of those is the `phoneNumber` data point which is typed as a string.

```typescript
const dataPoints = {
  ...,
  phoneNumber: {
    key: 'phoneNumber',
    valueType: 'string',
  }
} satisfies Record<string, DataPointDefinition>
```

Although phone numbers are indeed strings, we have an even more restrictive data point type in our system for phone strings. This data point type only allows for storing valid E.164 phone strings which is useful for any downstream action you want to perform with the phone number as you can be sure the string stored in a phone data point is a semantically correct phone number according to the E.164 format.

# To be situation

```typescript
const dataPoints = {
  ...,
  phoneNumber: {
    key: 'phoneNumber',
    valueType: 'telephone',
  }
} satisfies Record<string, DataPointDefinition>
```

Store the string we receive from Healthie in a phone data point. Given that we do not know that the string we are receiving from Healthie is in fact an E.164 valid phone string, we will have to apply some validation on our side to check whether the string we receive is a valid E.164 string or we try to parse it to one. This validation logic (validating incoming string to be a phone number) will come in handy for other actions as well, so I would create something in the `/lib` folder for that. Also see what is already in `lib/shared/validation`.

If the validation fails, I would store nothing (or `undefined`) in the phone data point and log an event for it:

```typescript
        await onComplete({
          data_points: {
            phoneNumber: undefined
          },
          events: [
            {
              date: new Date().toISOString(),
              text: { en: 'XYZ' },
              error: {
                category: 'XYZ',
                message: "Phone number from Healthie not stored because it isn't a valid E.164 phone number",
              },
            },
          ],
        })
```

# What's blocking

Extensions don't support ingesting data points with the type `phone` yet:

```typescript
export interface DataPointDefinition {
  key: string
  valueType: 'string' | 'number' | 'date' // add "| 'telephone'"
}
```

https://awellhealth.atlassian.net/browse/AST-5031

**Comment by bejoinka** [comment]
@michal-grzelak, can you check in with @nckhell tomorrow about this issue? I bring it up because the phone number is derived from an object, and I'm also not sure if anyone is using this datapoint from the extension today... meaning, "so what if we break backwards compatibility... if a 🌳 falls in the forest..."

**Comment by bejoinka** [comment]
Following up here. Please let me know if we can close this issue. I didn't see it referenced in a commit (e.g. `chore(): resolves #81`), so if it was resolved then github wouldn't have linked it or closed it out automatically. Thanks!
